acnportal/algorithms/preprocessing.py

```diff
@@ -178,3 +147,29 @@ def apply_minimum_charging_rate(
             session.min_rates[0] = 0
             session.max_rates[0] = 0
     return session_queue
+
+
+def remove_finished_sessions(active_sessions, infrastructure, period):
+    """ Remove any sessions where the remaining demand is less than threshold.
+
+    Args:
+        active_sessions (List[SessionInfo]): List of SessionInfo objects for
+            all active charging sessions.
+        infrastructure (InfrastructureInfo): Description of the charging
+            infrastructure.
+        period (int): Length of each time period in minutes.
```

Comment:
In general, can't the period be a float?

Reply:
I have never tried it. I think things may break, but I'm not sure. I believe that we make an implicit assumption that it divides 60 evenly in some places (which could still work for a float). But I would need to go over the code and make sure.